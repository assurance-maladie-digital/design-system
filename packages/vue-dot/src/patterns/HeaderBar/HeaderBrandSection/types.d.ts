export interface LogoInfo {
	src: string;
	alt: string;
}

export interface Service {
	mainTitle: string;
	highlight: string;
	subTitle: string;
}
