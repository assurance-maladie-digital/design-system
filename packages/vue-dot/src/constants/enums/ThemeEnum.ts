export enum ThemeEnum {
	DEFAULT = 'default',
	CNAM = 'cnam',
	AMELI = 'ameli',
	AMELI_PRO = 'ameli-pro',
	COMPTE_AMELI = 'compte-ameli',
	RISQUE_PRO = 'risque-pro',
	COMPTE_ENTREPRISE = 'compte-entreprise'
}

export const THEME_ENUM_VALUES = Object.values(ThemeEnum);
