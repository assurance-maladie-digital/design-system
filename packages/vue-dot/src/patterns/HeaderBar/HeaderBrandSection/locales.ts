export const locales = {
	logoCnam: 'Caisse nationale',
	logoAmeli: 'Ameli.fr',
	logoAmeliPro: 'AmeliPro'
};
