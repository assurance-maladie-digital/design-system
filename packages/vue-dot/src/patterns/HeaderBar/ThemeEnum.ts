export { ThemeEnum, THEME_ENUM_VALUES } from '../../constants/enums/ThemeEnum';
