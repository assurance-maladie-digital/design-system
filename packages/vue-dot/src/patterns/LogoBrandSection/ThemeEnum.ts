export enum ThemeEnum {
	DEFAULT = 'default',
	CNAM = 'cnam',
	AMELI = 'ameli',
	AMELI_PRO = 'ameli-pro',
	RISQUE_PRO = 'risque-pro',
	COMPTE_ENTREPRISE = 'compte-entreprise',
	COMPTE_AMELI = 'compte-ameli'
}

export const THEME_ENUM_VALUES = Object.values(ThemeEnum);
