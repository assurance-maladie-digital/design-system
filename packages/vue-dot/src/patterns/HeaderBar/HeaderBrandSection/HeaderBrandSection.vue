<template>
	<VSheet
		:height="height"
		class="vd-header-brand-section d-flex"
	>
		<component
			:is="logoContainerComponent"
			:aria-current-value="null"
			:aria-label="locales.homeLinkLabel"
			:to="homeLink"
			class="vd-header-home-link"
		>
			<Logo
				:hide-signature="hideSignature"
				:risque-pro="isRisquePro"
				:size="logoSize"
			/>
		</component>

		<slot>
			<svg
				v-if="showDivider"
				:width="dividerDimensions.width"
				:height="dividerDimensions.height"
				:fill="dividerColor"
				role="img"
				focusable="false"
				aria-hidden="true"
				xmlns="http://www.w3.org/2000/svg"
				viewBox="0 0 22 64"
			>
				<path d="M14.3 49.3c-.2 0-.4-.2-.4-.4V14.2c0-.2.2-.4.4-.4.3 0 .5.2.5.4v34.7c0 .2-.2.4-.5.4Z" />
			</svg>

			<img
				v-if="secondaryLogo"
				:src="secondaryLogo.src"
				:alt="secondaryLogo.alt"
			>

			<div
				v-else-if="service.title || service.subTitle"
				class="d-flex justify-center flex-column primary--text"
			>
				<h1
					v-if="service.title"
					:class="{ 'vd-compte-entreprise-title': isCompteEntreprise }"
					class="vd-header-title text-caption text-md-subtitle-1 font-weight-medium"
				>
					<template v-if="isCompteEntreprise">
						{{ service.title.text }}
						<span>{{ service.title.highlight }}</span>
					</template>

					<template v-else>
						{{ service.title }}
					</template>
				</h1>

				<h2
					v-if="showServiceSubTitle"
					class="vd-header-title text-caption"
				>
					{{ service.subTitle }}
				</h2>
			</div>
		</slot>
	</VSheet>
</template>

<script lang="ts">
	import Vue, { PropType } from 'vue';
	import Component, { mixins } from 'vue-class-component';

	import { tokens } from '@cnamts/design-tokens';

	import { Dimensions, Next } from '../../../types';
	import { LogoSizeEnum } from '../../../elements/Logo/LogoSizeEnum';

	import { ThemeEnum } from '../ThemeEnum';

	import { LogoInfo, Service } from './types';
	import { locales } from './locales';
	import { secondaryLogoMapping } from './secondaryLogoMapping';
	import { dividerDimensionsMapping } from './dividerDimensionsMapping';

	const Props = Vue.extend({
		props: {
			theme: {
				type: String as PropType<ThemeEnum>,
				required: true
			},
			serviceTitle: {
				type: String,
				default: undefined
			},
			serviceSubTitle: {
				type: String,
				default: undefined
			},
			mobileVersion: {
				type: Boolean,
				default: false
			},
			homeLink: {
				type: [String, Boolean, Object] as PropType<Next>,
				default: '/'
			}
		}
	});

	const MixinsDeclaration = mixins(Props);

	@Component
	export default class HeaderBrandSection extends MixinsDeclaration {
		locales = locales;

		get service(): Service {
			if (this.theme === ThemeEnum.COMPTE_ENTREPRISE) {
				const { title, subTitle } = locales.compteEntreprise;

				return {
					title,
					subTitle
				};
			}

			return {
				title: this.serviceTitle,
				subTitle: this.serviceSubTitle
			};
		}

		get height(): string {
			if (this.mobileVersion && this.hasSecondaryLogo) {
				return '32px';
			}

			return this.mobileVersion ? '40px' : '64px';
		}

		get isRisquePro(): boolean {
			return this.theme === ThemeEnum.RISQUE_PRO;
		}

		get isCompteEntreprise(): boolean {
			return this.theme === ThemeEnum.COMPTE_ENTREPRISE;
		}

		get hideSignature(): boolean {
			if (this.theme === ThemeEnum.COMPTE_ENTREPRISE) {
				return true;
			}

			return Boolean(this.$slots.default);
		}

		get secondaryLogo(): LogoInfo | undefined {
			return secondaryLogoMapping[this.theme];
		}

		get hasSecondaryLogo(): boolean {
			return Boolean(this.$slots.default || this.secondaryLogo);
		}

		get logoContainerComponent(): string {
			return this.homeLink ? 'RouterLink' : 'div';
		}

		get showDivider(): boolean {
			return Boolean(this.hasSecondaryLogo || this.service.title);
		}

		get showServiceSubTitle(): boolean {
			return Boolean(this.service.title && this.service.subTitle && !this.mobileVersion);
		}

		get dividerColor(): string {
			switch (this.theme) {
				case ThemeEnum.CNAM:
				case ThemeEnum.AMELI_PRO: {
					return tokens.colors.secondary;
				}

				case ThemeEnum.COMPTE_ENTREPRISE: {
					return '#cd545b'; // Brique 100 color
				}

				default: {
					return tokens.colors.primary;
				}
			}
		}

		get dividerDimensions(): Dimensions {
			const { xSmall, small, normal } = dividerDimensionsMapping;

			if (this.mobileVersion && this.hasSecondaryLogo) {
				return xSmall;
			}

			if (this.mobileVersion) {
				return small;
			}

			return normal;
		}

		get logoSize(): LogoSizeEnum {
			if (this.mobileVersion && this.hasSecondaryLogo) {
				return LogoSizeEnum.X_SMALL;
			}

			if (this.mobileVersion) {
				return LogoSizeEnum.SMALL;
			}

			return LogoSizeEnum.NORMAL;
		}
	}
</script>

<style lang="scss" scoped>
	.vd-header-brand-section {
		overflow: hidden;

		.vd-header-title {
			line-height: 1.45 !important;
		}

		.vd-compte-entreprise-title {
			font-weight: 700 !important;

			span {
				color: #cd545b;
			}
		}

		img,
		::v-deep img {
			width: auto;
			height: 100%;
			flex: none;
		}

		svg,
		.vd-header-home-link {
			flex: none;
		}
	}
</style>
