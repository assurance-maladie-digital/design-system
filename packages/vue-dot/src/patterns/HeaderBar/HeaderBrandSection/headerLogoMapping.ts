import { IndexedObject } from 'src/types';
import { locales } from './locales';
import { LogoInfo } from './types';

export const headerLogoMapping: IndexedObject<LogoInfo> = {
	logoAM: {
		src: require('../../../assets/logos/desktop/assurance-maladie.svg'),
		alt: locales.logoAssuranceMaladie
	},
	logoRisquePro: {
		src: require('../../../assets/logos/desktop/risque-pro.svg'),
		alt: locales.logoRisquePro
	}
};

export const secondaryLogoMapping: IndexedObject<LogoInfo> = {
	ameli: {
		src: require('../../../assets/logos/desktop/ameli.svg'),
		alt: locales.logoAmeli
	},
	ameliPro: {
		src: require('../../../assets/logos/desktop/ameli-pro.svg'),
		alt: locales.logoAmeliPro
	}
};
