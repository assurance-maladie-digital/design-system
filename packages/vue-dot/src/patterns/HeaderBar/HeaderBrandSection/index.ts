import HeaderBrandSection from './HeaderBrandSection.vue';

export default HeaderBrandSection;
