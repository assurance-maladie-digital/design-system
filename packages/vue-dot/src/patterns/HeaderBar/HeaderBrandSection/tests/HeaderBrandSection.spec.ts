import Vue from 'vue';
import { Wrapper } from '@vue/test-utils';

import { mountComponent } from '@/tests/unit';
import { html } from '@/tests/utils/html';

import HeaderBrandSection from '../';
import { ThemeEnum } from '../../ThemeEnum';

import Logo from '../../../../elements/Logo';

Vue.component('Logo', Logo);

let wrapper: Wrapper<Vue>;

// Tests
describe('HeaderBrandSection', () => {
	it('renders correctly', () => {
		// Mount component
		wrapper = mountComponent(HeaderBrandSection, {
			propsData: {
				theme: ThemeEnum.DEFAULT
			}
		});

		expect(html(wrapper)).toMatchSnapshot();
	});
});
