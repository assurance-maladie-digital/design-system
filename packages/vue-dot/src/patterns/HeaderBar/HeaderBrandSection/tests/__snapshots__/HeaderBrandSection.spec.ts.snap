// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBrandSection renders correctly 1`] = `
<vsheet-stub height="64px" tag="div" class="vd-header-brand-section d-flex">
  <routerlink-stub to="/">
    <logo-stub size="normal"></logo-stub>
  </routerlink-stub>
  <!---->
  <!---->
</vsheet-stub>
`;
